hw/xbox/xid-sdl.c

```diff
@@ -464,6 +479,66 @@ static void usb_xid_class_initfn(ObjectClass *klass, void *data)
     uc->handle_attach  = usb_desc_attach;
 }
 
+static HapticEmulationMode select_haptic_emulation_mode(SDL_Haptic *haptic)
+{
+    uint query_result = SDL_HapticQuery(haptic);
```

Comment:
The type `uint` is undefined and causing compilation failures:

```C:/projects/xqemu/hw/xbox/xid-sdl.c:484:5: error: unknown type name 'uint'; did you mean 'guint'?```

We should probably follow [the docs](https://wiki.libsdl.org/SDL_HapticQuery) and just use `unsigned int` here.